src/components/LayerMetadataPopupService.js

```diff
@@ -18,29 +18,36 @@ goog.require('ga_wms_service');
         gaMapUtils, gaWms) {
       var popupContent = '<div ng-bind-html="options.result.html"></div>';
 
+      // Called to update the content
+      var updateContent = function(popup, layer) {
+        var promise;
+        if (layer.bodId) {
+          promise = gaLayers.getMetaDataOfLayer(layer.bodId);
+        } else if (gaMapUtils.isExternalWmsLayer(layer)) {
+          promise = gaWms.getLegend(layer);
+        }
+        return promise.then(function(resp) {
+          popup.scope.options.result.html = $sce.trustAsHtml(resp.data);
```

Comment:
here you should set also the lang to the popup options:

`popup.scope.options.lang = newLang;`

Otherwise if there is an error in the request when you will reopen the popup, the request will not be sent.